applications/ConvectionDiffusionApplication/tests/bfecc_convection_test.py

```diff
@@ -100,34 +112,47 @@ def runTest(self):
-                    reference_file.close()
+                json_output_settings = KratosMultiphysics.Parameters(r'''{
+                        "output_variables": ["TEMPERATURE"],
+                        "output_file_name": "",
+                        "model_part_name": "MainModelPart",
+                        "time_frequency": 0.0
+                }''')
+                json_output_settings["output_file_name"].SetString(self.reference_file + "_results.json")
+                json_output_process = JsonOutputProcess(self.model, json_output_settings)
+                json_output_process.ExecuteInitialize()
+                json_output_process.ExecuteBeforeSolutionLoop()
+                json_output_process.ExecuteFinalizeSolutionStep()
             else:
-                with open(self.reference_file + '_results' + '.csv','w') as reference_file:
-                    reference_file.write("#ID, TEMPERATURE\n")
-                    for node in self.main_model_part.Nodes:
-                        reference_file.write("{0}, {1}\n".format(node.Id, node.GetSolutionStepValue(KratosMultiphysics.TEMPERATURE, 0)))
-                    reference_file.close()
-
-                compare_files_settings = KratosMultiphysics.Parameters(r'''{
-                    "reference_file_name"   : "bfecc_convection_test.csv",
-                    "output_file_name"      : "bfecc_convection_test_results.csv",
-                    "remove_output_file"    : true,
-                    "comparison_type"       : "deterministic"
+                json_check_parameters = KratosMultiphysics.Parameters(r'''{
+                    "check_variables"      : ["TEMPERATURE"],
+                    "input_file_name"      : "",
+                    "model_part_name"      : "MainModelPart",
+                    "time_frequency"       : 0.0
                 }''')
-                CompareTwoFilesCheckProcess(compare_files_settings).Execute()
+                json_check_parameters["input_file_name"].SetString(self.reference_file + "_results.json")
+                json_check_process = FromJsonCheckResultProcess(self.model, json_check_parameters)
+                json_check_process.ExecuteInitialize()
+                json_check_process.ExecuteBeforeSolutionLoop()
+                json_check_process.ExecuteFinalizeSolutionStep()
 
     def testBFECCConvection(self):
+        self.has_limiter = False
+        self.reference_file = "bfecc_convection_test"
+        self.setUp()
+        self.runTest()
+        self.checkResults()
+        self.tearDown()
+
+    def testBFECCElementalLimiterConvection(self):
+        self.has_limiter = True
+        self.reference_file = "bfecc_elemental_limiter_convection_test"
         self.setUp()
         self.runTest()
         self.checkResults()
         self.tearDown()
 
 if __name__ == '__main__':
     test = BFECCConvectionTest()
-    test.setUp()
     test.testBFECCConvection()
-    test.tearDown()
+    # test.testBFECCElementalLimiterConvection()
```

Comment:
intentionally commented?

Reply:
Yes. To make it easier to change between the two tests if the entire suite is not run.